R/viz.R

```diff
@@ -4522,18 +4548,19 @@ fancy_qc_plot = function(these_samples,
     this_meta = metadata
   }
 
-  #filter metadata on selected cohort (if filter_cohort is called)
-  if(missing(these_samples) && !missing(filter_cohort)){
-    these_samples = dplyr::filter(this_meta, cohort == filter_cohort) %>%
-      dplyr::select(sample_id) %>%
-      as.data.frame()
-  }
-
-  #filter metadata on pathology (if filter_pathology is called)
-  if(missing(these_samples) && !missing(filter_pathology)){
-    these_samples = dplyr::filter(this_meta, pathology == filter_pathology) %>%
-      dplyr::select(sample_id) %>%
-      as.data.frame()
+  if(missing(these_samples)){
```

Comment:
Same issue as above. Is the idea here that someone will only ever be expected to filter using one or the other but never combine them? If so, then you will need to check that they never provide more than one filter and throw an error if they do

Reply:
Yes, see my response to your previous comment. It's the same for all these functions. I think it's best to have the function filter on both if both parameters are used.